crawler.js

```diff
@@ -1333,19 +1333,29 @@ self.__bx_behaviors.selectMainBehavior();
     }
   }
 
-  async queueUrl(seedId, url, depth, extraHops = 0) {
+  async queueUrl(seedId, url, depth, extraHops, logDetails = {}) {
     if (this.limitHit) {
       return false;
     }
 
-    if (!await this.crawlState.addToQueue({url, seedId, depth, extraHops}, this.pageLimit)) {
+    const result = await this.crawlState.addToQueue({url, seedId, depth, extraHops}, this.pageLimit);
+
+    switch (result) {
+    case QueueState.ADDED:
+      logger.debug("Queued new page url", {url, ...logDetails}, "links");
+      return true;
+
+    case QueueState.LIMIT_HIT:
+      logger.debug("Not queued page url, at page limit", {url, ...logDetails}, "links");
       this.limitHit = true;
       return false;
-    } else {
-      logger.debug(`Queued url ${url}`);
+
+    case QueueState.DUPE_URL:
+      logger.debug("Not queued page url, already seen", {url, ...logDetails}, "links");
+      return false;
```

Comment:
Cases should be indented by 2

Reply:
Our lint settings actually format it this way currently, maybe something to take a look at in a separate pass.

Reply:
Ah okay, noted!